File: src/main/java/base/mvp/BaseContract.java
package base.mvp;

import android.databinding.Observable;

public interface BaseContract {

    interface View<P extends Presenter, VM extends ViewModel> {

        void setPresenter(P presenter);

        void setViewModel(VM viewModel);

    }

    interface Presenter {

        void start();

        void resume();

        void pause();

        void stop();

    }

    interface ViewModel extends Observable {

        void onPropertyChanged(int propertyId, Runnable action);
    }
}

File: src/main/java/base/mvp/BaseContract.kt
package base.mvp

import android.databinding.Observable

interface BaseContract {

    interface Presenter {

        fun start()

        fun resume()

        fun pause()

        fun stop()

    }

    interface ViewModel : Observable {

        fun onPropertyChanged(propertyId: Int, action: () -> Unit)
    }
}
